src/components/PromptModal.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+
+interface PromptModalProps {
+  open: boolean;
+  message: React.ReactNode;
+  value: string;
+  onChange: (next: string) => void;
+  confirmLabel?: string;
+  cancelLabel?: string;
+  onConfirm: () => void;
+  onCancel: () => void;
+  confirmDisabled?: boolean;
```

Comment:
Is there potential to apply DRY and pull confirmation stuff out into a separate reusable component?